packages/modules/product/src/models/product-image.ts

```diff
@@ -58,17 +58,34 @@ class ProductImage {
   @Property({ columnType: "timestamptz", nullable: true })
   deleted_at?: Date
 
-  @ManyToMany(() => Product, (product) => product.images)
-  products = new Collection<Product>(this)
+  @Property({ columnType: "integer", default: 0 })
+  rank: number
+
+  @ManyToOne(() => Product, {
+    columnType: "text",
+    nullable: true,
```

Comment:
thought: Should this be not null?

Reply:
yes we can remove the nullable, plus with the onDelete cascade it can't happen that the image exists after a product deletion.

Reply:
If I remove nullable, creating an image fails with: `ValidationError: Value for ProductImage.product_id is required`

Reply:
You might need to assign the `product_id` directly to the image upon creating the product, similar to what you added in the update path.

Reply:
Yes we can do that, but then I will need to refetch the products before we return from both functions, because otherwise the images won't be present (breaking change and breaks a bunch of pre-existing tests). I can just do that, but just want to check if thats alright since we are adding both more separate DB writes and an additional DB read.

Reply:
Not sure I understand, where would you need to refetch products?

Reply:
I can do that, but then I will also need to overwrite the ProductService, since it has a bunch of tests for creating/updating/restoring products with images and they start failing. I am assuming it's the same reason for why options and variants have the field as nullable.

Reply:
@adrien2p could you maybe point me to where in the code we create the MedusaInternalServices so I can get an understanding of what I am overwriting, it's a bit hard to gage from the types of the .create, update, upsert, and upsertAndReplace functions.